### PR TITLE
JavaScript capstone project - Your API-based webapp

### PR DESCRIPTION
**Pull request by afcastaneda223**
# Documentation

All the documentation is on the README file.

[Video Link](https://drive.google.com/file/d/1yJk0YRkPb11ETjitphvXPAH80JEg-WQo/view?usp=sharing)

## Updates:

- Display a list of items on the Homepage.
- Create feature: add new like.
- Display the number of likes for each item on the Homepage.
- Add all items counter on the Homepage.
- Add tests for items counter.
- Display Comments pop up with selected item's details 
- Display comments for a given item on the Comments
- Add tests for comments counter 
- Create feature: add a new comment 
- Add comments counter 

## Notes:

Coded by @afcastaneda223 & @the-catalystmc 
✅  - No linters errors were detected.

### CODE REVIEW

File: src/itemCounter.test.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  const mealLenght = data.meals.lenght;
+  const itemCountTest = await itemCount.countMeals();
+
+  expect(itemCountTest).toBe(mealLenght);
+});
+
+test('the array has elements it cant return 0 ', async () => {
+  const itemCountTest = await itemCount.countMeals();
+  expect(itemCountTest).not.toBe(0);
+});
```
Comment (mricanho) [inline src/itemCounter.test.js]: - [ ] This test is throwing an error, please double-check it and if in your local environment this test pass, please add a screenshot and the next code reviewer will look further into it.

Test failing:

image

**Review by mricanho — CHANGES_REQUESTED** [review]
Hi @afcastaneda223 @the-catalystmc, 

You really have an excellent capstone here, you followed the gitflow, the mechanics around the reviews(Although I'll wish you try more frequent PRs, yet two are acceptable), you used ES6 and your JS code is clean and easy to follow, good job on your API, it works as expected, the same for the likes and popup comment; good use of bootstrap for the style 🔥🔥

There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️

- [ ] You are using a file with HTML, so you need to set up the HTML's linter, [follow this link](https://github.com/microverseinc/linters-config/tree/master/html-css-js) and add it, please.

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

almostthere

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/itemCounter.test.js (hunk 1 above)
Comment (afcastaneda223) [reply src/itemCounter.test.js]: Thank you for the review, yes on my local environment the test for itemCounter.test.js work perfectly as shown on the next screenshot:
image

Comment (the-catalystmc) [reply src/itemCounter.test.js]: I messaged you on Slack

**Review by hemantso — APPROVED** [review]
Hi @afcastaneda223 and @the-catalystmc, 

Your application is working absolutely fine. Good work on meeting all requirements. 🚀 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

## Congratulations! 🎉 

Alt Text

Cheers and happy coding 😃

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. You can mention me (@hemant-soni-vst-au4) so I can see your comments and respond.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**